src/main/java/Daniela_Chihai/com/Endava/SoA5/Java_Collections/User.java

```diff
@@ -0,0 +1,39 @@
+package Daniela_Chihai.com.Endava.SoA5.Java_Collections;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class User {
+    private long id;
+    private String username;
+    private Map<String, Address> addresses;
+
+    public User(long id, String username, Map<String, Address> addresses) {
+        this.id = id;
+        this.username = username;
+        this.addresses = addresses;
+    }
+
+    @Override
+    public String toString() {
+        return "User{" +
+                "id=" + id +
+                ", username='" + username + '\'' +
+                ", addresses=" + addresses +
+                '}';
+    }
+
+    public List<User> filterUserByCity(List<User> users, String city) {
```

Comment:
Please divide this method into 2 one will return boolean and reuse that result in the second method